rules/private/phases/phase_semanticdb.bzl

```diff
@@ -0,0 +1,42 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@rules_scala_annex//rules:providers.bzl", _ScalaConfiguration = "ScalaConfiguration")
+
+#
+# PHASE: semanticdb
+#
+# Configures the compiler to output SemanticDB metadata. Note that this phase won't work without the
+# SemanticDB compiler plugin being enabled.
+#
+def phase_semanticdb(ctx, g):
+    scala_configuration = ctx.attr.scala[_ScalaConfiguration]
+
+    if scala_configuration.semanticdb_bundle:
+        return struct(outputs = [], scalacopts = [])
+
+    outputs = []
+    semanticdb_directory = paths.join("_semanticdb/", ctx.label.name)
+    semanticdb_target_root = paths.join(paths.dirname(ctx.outputs.jar.path), semanticdb_directory)
+
+    for source in ctx.files.srcs:
+        if source.extension == "scala":
+            output_filename = paths.join(
+                semanticdb_directory,
+                "META-INF",
+                "semanticdb",
+                "{}.semanticdb".format(source.path),
+            )
+
+            outputs.append(ctx.actions.declare_file(output_filename))
+
+    if scala_configuration.version.startswith("2"):
```

Comment:
I'm not sure how much we care about this, but how does this work for things like Scalajs or Scala Native?

Reply:
Hmm, it depends on what `version` is set to. The only other place in which we use it is `phase_bootstrap_compile`, where we also check if we're dealing with Scala 2 or 3.

Now that I think about it, is SemanticDB even a concept in Scala.js and Scala Native? I think both Scala.js and Scala Native use the same compiler, so it should be. For now, I'll leave this as is and dig into it further if folks want Scala.js and/or Scala Native support (assuming that's ok with you).